extractor/src/main/java/org/schabi/newpipe/extractor/stream/Stream.java

```diff
@@ -1,108 +1,191 @@
+     * @param content        the content or URL, depending on whether isUrl is true
+     * @param isUrl          whether content is the URL or the actual content of e.g. a DASH
+     *                       manifest
+     * @param format         the {@link MediaFormat}, which can be null
+     * @param deliveryMethod the delivery method
+     * @param baseUrl        the base URL of the content if the stream is a DASH or an HLS
+     *                       manifest, which can be null
      */
-    public Stream(String url, MediaFormat format) {
-        this(url, null, format);
+    protected Stream(final String id,
+                     final String content,
+                     final boolean isUrl,
+                     @Nullable final MediaFormat format,
+                     final DeliveryMethod deliveryMethod,
+                     @Nullable final String baseUrl) {
+        this.id = id;
+        this.content = content;
+        this.isUrl = isUrl;
+        this.mediaFormat = format;
+        this.deliveryMethod = deliveryMethod;
+        this.baseUrl = baseUrl;
     }
 
     /**
-     * Instantiates a new stream object.
+     * Check if the list already contains one stream with equals stats.
      *
-     * @param url        the url
-     * @param torrentUrl the url to torrent file, example https://webtorrent.io/torrents/big-buck-bunny.torrent
-     * @param format     the format
+     * @param stream the stream which will be compared to the streams in the stream list
+     * @param streamList the list of {@link Stream Streams} which will be compared
      */
-    public Stream(String url, String torrentUrl, MediaFormat format) {
-        this.url = url;
-        this.torrentUrl = torrentUrl;
-        this.format = format.id;
-        this.mediaFormat = format;
+    public static boolean containSimilarStream(final Stream stream,
+                                               final List<? extends Stream> streamList) {
+        if (isNullOrEmpty(streamList)) return false;
+        for (final Stream cmpStream : streamList) {
+            if (stream.equalStats(cmpStream)) {
+                return true;
+            }
+        }
+        return false;
     }
 
     /**
-     * Reveals whether two streams have the same stats (format and bitrate, for example)
+     * Reveals whether two streams have the same stats (format and bitrate, for example).
+     *
+     * <p>
+     * It returns always false if the id of the stream passed is unknown (or also if the stream
+     * passed is null) or if the format of the stream compared is not known.
+     * </p>
+     * @param cmp the stream object to be compared to this stream object
      */
-    public boolean equalStats(Stream cmp) {
-        return cmp != null && getFormatId() == cmp.getFormatId();
+    public boolean equalStats(final Stream cmp) {
+        if (mediaFormat != null && cmp != null && cmp.getFormat() != null) {
+            return mediaFormat.id == cmp.getFormat().id;
```

Comment:
You can access private fields of instances different than `this` ;-)
```suggestion
        if (mediaFormat != null && cmp != null && cmp.mediaFormat != null) {
            return mediaFormat.id == cmp.mediaFormat.id;
```

Reply:
Oh, I didn't know this. Thanks!